Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -896,8 +900,11 @@ class BrowserViewController: UIViewController {
 
         if isAboutHomeURL {
             showFirefoxHome(inline: true)
-
-            if focusUrlBar {
+            
+            if userHasPressedHomeButton {
+                userHasPressedHomeButton = false
+                
+            } else if focusUrlBar {
                 if let viewcontroller = presentedViewController as? OnViewDismissable {
```

Comment:
nit: the following could have it's own private func? like `private func focusUrlBar()` or something. Just wanna try reducing the amount of arrow code here.